shm/src/map.cpp

```diff
@@ -10,7 +10,13 @@ constexpr size_t kMapWidth = 50;
 constexpr size_t kMapHeight = 50;
 
 Map::Map() {
-    generateIslands();
+    std::random_device rd;
+    std::mt19937 eng(rd());
+    std::uniform_int_distribution<> distribX(0, kMapWidth - 1);
+    std::uniform_int_distribution<> distribY(0, kMapHeight - 1);
+
+    islands_.resize(kIslandAmt);
+    std::generate(islands_.begin(), islands_.end(), [&]() { return Island(distribX(eng), distribY(eng)); });
```

Comment:
Are you sure that this won't generate Islands with the same coordinates?

Reply:
I didn't think about it. Ok I have another idea. That I will try out :)

Reply:
I've proposed solution with uniform_int_distribution but our team chose vector approach :)

Reply:
```suggestion
    std::random_device rd;
    std::mt19937 eng(rd());
    islands_.reserve(kIslandAmt);

    std::uniform_int_distribution<> widthGen(0, kMapWidth);
    std::uniform_int_distribution<> heightGen(0, kMapHeight);

    std::set<std::pair<int, int>> uniqueIslands;
    int x = 0;
    int y = 0;
    while (unique_islands.size() < kIslandAmt) {
        x = widthGen(eng);
        y = heightGen(eng);

        uniqueIslands.insert({x, y});
    }
    std::transform(uniqueIslands.begin(), uniqueIslands.end(),
                   std::back_inserter(islands_),
                   [](const auto& coords) {
                       return Island(coords.first, coords.second);
                   });
```
That was my proposition ;)

Reply:
Anyway I am not sure if we have to follow every suggestion from PR in coders-school :)